homeassistant/components/hive/manifest.json

```diff
@@ -3,7 +3,7 @@
   "name": "Hive",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/hive",
-  "requirements": ["pyhiveapi==0.4.2"],
+  "requirements": ["pyhiveapi==0.5.3"],
```

Comment:
Why do we need to bump the library to introduce entity descriptions and categories?

Reply:
@MartinHjelmare the pyhiveapi library has multiple use cases other than just home assistant. It is actively being developed with new functionality and features including performance improvements. By bumping up the library home assistant may not be taking advantage of new features yet, but it will take advantage of the performance improvments.

An example of this would be the migration to using the boto3 library in pyhiveapi for authentication instead of the hive api, as it not only gives us better performance it also give us better management of authentication errors.

The change required for this PR in pyhiveapi is around changing a flag on some hive devices so that configuration entities and diagnostic entities are now sent to home assistant and linked to the device.

Reply:
But we don't need to do that in this PR, right? That can be a separate PR. This PR should just refactor existing code. Please separate different kinds of changes into different PRs.

Reply:
@MartinHjelmare we do need this in this PR  for the updated config flags.

Reply:
Which flags are that?

Reply:
@MartinHjelmare In the pyhiveapi code there are some custom flags which are now set too false which allows them to be included in the devices/entities returned to home assistant. in the current version of the code the custom flag is set too True so they are not returned. Ultimately within this release we have moved the control of what entities are shown into the home assistant code base, using the entity descriptions instead of using the pyhiveapi custom flag. In a future pyhiveapi release I envisage I can remove that custom flag completely. Below is the pyhiveapi config I am referring too.

https://github.com/Pyhass/Pyhiveapi/blob/4e09ab78ac2c24810b74aa6774c32e18267be8fa/pyhiveapi/apyhiveapi/helper/const.py#L77-L167

Reply:
Ok, so this PR should only refactor the existing code to use entity descriptions and not add any new features. Then create another PR, after this PR is merged, that bumps the library and exposes the new features as entities.

Reply:
I’m really confused now are you asking we have 2 active versions of pyhiveapi? I don’t think that’s a realistic option or understand why pyhiveapi has to wait for home assistant to be ready before it can release new features. The current code in this PR has been tested locally and it works with the latest version, so I don’t understand why it needs a separate PR and 2 active versions to be maintained.

Reply:
We don't need to different versions of the library.

I'm just asking for smaller PRs with a separate set of changes in each one instead of one big PR with different kinds of changes.

In this PR, at the moment, you're both doing a refactor and adding new features. The refactor consists of using entity descriptions and `_attr_` attributes and consolidating entity properties to a single base class. The new features consists of new kinds of entities and a library bump.

We should split that work in two different PRs. The first PR should do the refactor and the second PR should add the new features including a library bump.

Reply:
So if I create a new PR just for the library bump get that approved and then continue with this pull request?

Reply:
The main problem is that this PR is mixing a refactor with new features. The library bump is part of the new features but there are other changes here that are also new features, that should be split out.

My suggestion is to remove all the new features from this PR. After we merge here you can open another PR with the new features.

Reply:
Ok I will remove the library bump from this PR my next PR is going to be implementing hive authentication features and fixing an outstanding issue that’s been raised where users at the minute cannot login and use the integration would the PR to include the library bump?

Reply:
Fixes should preferably be separate from other new features. It's of course ok to bump the library in the same PR if that's required for the fix, but in general you should be aiming at as few changes as possible in each PR.

Think about what the thing is that you want to achieve with each PR. Each PR should just do one thing and contain the minimum set of changes to achieve that.

Reply:
@MartinHjelmare  the library bump has been removed now